inference-engine/src/transformations/src/transformations/convert_divide.cpp

```diff
@@ -17,7 +17,8 @@ void ngraph::pass::ConvertDivide::convert_divide() {
 
     ngraph::graph_rewrite_callback callback = [](pattern::Matcher& m) {
         auto div = std::dynamic_pointer_cast<ngraph::opset1::Divide> (m.get_match_root());
-        if (!div) {
+        // We can not apply this transformation in case with integer input data type
+        if (!div || div->input(0).get_element_type().is_integral()) {
```

Comment:
Is `is_integral` true for `undefined` and `dynamic`?

Reply:
is_integral = !is_real I think it works for case with dynamic. And there is no undefined type

Reply:
What do you mean by "works for case with dynamic"?
There is definetly such a type - it's defined element_type.hpp:42.

I mean semantically, we cannot do such transformation if type is dynamic, because we don't know if it's integral or not, so I guess check on static can be introduced to be more robust (I guess if there is a case we will support dynamic types it'll be difficult to debug such behavior)

Reply:
But let's leave it as is in the request to not block Faster debugging/developemt